File: app/src/main/java/org/jaaksi/pickerview/demo/TestPickerViewFragment.java
package org.jaaksi.pickerview.demo;

import android.graphics.Color;
import android.view.View;
import org.jaaksi.pickerview.adapter.NumericWheelAdapter;
import org.jaaksi.pickerview.util.Util;
import org.jaaksi.pickerview.widget.BasePickerView;
import org.jaaksi.pickerview.widget.DefaultCenterDecoration;
import org.jaaksi.pickerview.widget.PickerView;

/**
 * Created by fuchaoyang on 2018/2/5.<br/>
 * description：
 */

public class TestPickerViewFragment extends BaseFragment {
  PickerView<Integer> mPickerView;

  @Override
  protected int getLayoutId() {
    return R.layout.fragment_test_picker;
  }

  @Override
  protected void initView(View view) {
    mPickerView = view.findViewById(R.id.pickerview);
    mPickerView.setAdapter(new NumericWheelAdapter(1, 10));
    // 覆盖xml中的水平方向
    mPickerView.setHorizontal(false);
    mPickerView.setTextSize(15, 22);
    mPickerView.setIsCirculation(true);
    //mPickerView.setAlignment(Layout.Alignment.ALIGN_CENTER);
    mPickerView.setCanTap(false);
    mPickerView.setDisallowInterceptTouch(false);
    // 覆盖xml设置的7
    mPickerView.setVisibleItemCount(5);
    mPickerView.setItemSize(50);
    // 格式化内容
    mPickerView.setFormatter(new BasePickerView.Formatter() {
      @Override
      public CharSequence format(BasePickerView pickerView, int position,
        CharSequence charSequence) {
        return charSequence + "万年";
      }
    });
    int margin = Util.dip2px(mActivity, 5);
    DefaultCenterDecoration centerDecoration =
      new DefaultCenterDecoration(getActivity()).setLineColor(Color.GREEN)
        .setMargin(margin, -margin, margin, -margin)
        .setLineWidth(3)
        .setDrawable(Color.RED);
    mPickerView.setCenterDecoration(centerDecoration);
    //mPickerView.setSelectedPosition(1);
    // 设置centerPosition
    mPickerView.postDelayed(new Runnable() {
      @Override
      public void run() {
        mPickerView.setCenterPosition(1);
      }
    }, 1000);
  }
}

File: app/src/main/java/org/jaaksi/pickerview/demo/ui/TestPickerViewFragment.kt
package org.jaaksi.pickerview.demo.ui

import android.os.Bundle
import androidx.core.graphics.toColorInt
import org.jaaksi.pickerview.adapter.NumericWheelAdapter
import org.jaaksi.pickerview.demo.base.BaseFragment
import org.jaaksi.pickerview.demo.databinding.FragmentTestPickerBinding
import org.jaaksi.pickerview.picker.TimePicker
import org.jaaksi.pickerview.util.DateUtil
import org.jaaksi.pickerview.util.Util
import org.jaaksi.pickerview.widget.BasePickerView
import org.jaaksi.pickerview.widget.DefaultCenterDecoration

/**
 * Created by fuchaoyang on 2018/2/5.<br></br>
 * description：
 */
class TestPickerViewFragment : BaseFragment<FragmentTestPickerBinding>() {

    override fun initView(savedInstanceState: Bundle?) {
        super.initView(savedInstanceState)
        binding.pickerview.adapter = NumericWheelAdapter(1, 9)
        binding.pickerview.apply {
            // 覆盖xml中的水平方向
            isHorizontal = false
            setTextSize(18, 20)
            setIsCirculation(true)
            //setAlignment(Layout.Alignment.ALIGN_CENTER);
            isCanTap = false
            isDisallowInterceptTouch = false
            // 覆盖xml设置的7
            visibleItemCount = 5
            // 格式化内容
            formatter =
                BasePickerView.Formatter { pickerView, position, charSequence -> charSequence.toString() + "years" }

            val margin = Util.dip2px(requireContext(), 5f)
            val centerDecoration =
                DefaultCenterDecoration(requireContext())
                    .setLineColor("#1A0000FF".toColorInt())
                    //.setMargin(margin, -margin, margin, -margin)
                    .setLineWidth(3f)
                    .setDrawable("#1AF84055".toColorInt())
            setCenterDecoration(centerDecoration)
            // 设置centerPosition
            //postDelayed(1000) {
            //    centerPosition = 1
            //}

            // TimePicker用在View中
            TimePicker.Builder(
                requireContext(), TimePicker.TYPE_ALL
            ) { picker, date ->
                binding.tvTime.text = DateUtil.createDateFormat("yyyy年MM月dd日 HH:mm").format(date)
            } // 设置时间区间
                .dialog(null)
                .setRangDate(
                    1540361760000L,
                    System.currentTimeMillis()
                ) //.setContainsStarDate(true)
                //.setTimeMinuteOffset(10)
                // 设置选中时间
                .setSelectedDate(System.currentTimeMillis())
                // 设置pickerview样式
                .setInterceptor { pickerView, params ->
                    pickerView.visibleItemCount = 5
                    // 将年月设置为循环的
                    val type = pickerView.tag as Int
                    //if (type == TimePicker.TYPE_YEAR || type == TimePicker.TYPE_MONTH) {
                    //  pickerView.setIsCirculation(true);
                    //}
                }.create().view().let {
                    binding.timepickerLayout.addView(it)
                }
        }

    }

}